packages/component/src/provider.ts

```diff
@@ -261,11 +261,18 @@ export class Provider<T extends Name> {
    * @param callback - a function that will be invoked  after the provider has been initialized by calling provider.initialize().
    * The function is invoked SYNCHRONOUSLY, so it should not execute any longrunning tasks in order to not block the program.
    *
+   * @param identifier An optional instance identifier
    * @returns a function to unregister the callback
    */
-  onInit(callback: OnInitCallBack<T>): () => void {
+  onInit(callback: OnInitCallBack<T>, identifier?: string): () => void {
     this.onInitCallbacks.add(callback);
 
+    const instance = this.getImmediate({ identifier, optional: true });
```

Comment:
Can you please add a comment that OnInit() is only intended to be used for `EXPLICIT` components, so getImmediate() will never initialize a new instance?